Tests/FunctionalTests/TestDiscoveryTests.swift

```diff
@@ -12,159 +12,166 @@
 
 import Basics
 import PackageModel
+import SPMBuildCore
 import _InternalTestSupport
-import XCTest
+import Testing
+import struct Foundation.UUID
+import class Foundation.ProcessInfo
 
-final class TestDiscoveryTests: XCTestCase {
-    func testBuild() async throws {
+@Suite
+struct TestDiscoveryTests {
+
+    // Conditional parameters to tests are not supported by Swift Testing currently, so we construct the list of params here.
+    static var buildSystems: [BuildSystemProvider.Kind] = {
+        if ProcessInfo.hostOperatingSystem == .windows {
+            // Windows currently fails due to long path handling issues on Windows
+            return [BuildSystemProvider.Kind.native]
```

Comment:
issue (blocking): we should execute these tests on Windows and mark each test as `withKnownIssue` so we get signal, so we can ensure the test is enabled once the underlying windows path issue is fixed.

Have a looks at `Tests/FunctionalTests/TraitTests.swift` for how those tests handle it.